src/uwtools/api/driver.py

```diff
@@ -4,6 +4,103 @@
+
+from uwtools.drivers.support import graph
+from uwtools.drivers.support import tasks as _tasks
+from uwtools.logging import log
+from uwtools.utils.api import ensure_data_source
+
+
+def execute(  # pylint: disable=unused-argument
+    module: str,
+    classname: str,
+    task: str,
+    schema_file: str,
+    module_dir: Optional[Union[Path, str]] = None,
+    config: Optional[Union[Path, str]] = None,
+    cycle: Optional[datetime] = None,
+    leadtime: Optional[timedelta] = None,
+    batch: Optional[bool] = False,
+    dry_run: Optional[bool] = False,
+    graph_file: Optional[Union[Path, str]] = None,
+    key_path: Optional[list[str]] = None,
+    stdin_ok: Optional[bool] = False,
+) -> bool:
+    """
+    Execute a task.
+
+    If ``batch`` is specified, a runscript will be written and submitted to the batch system.
+    Otherwise, the executable will be run directly on the current system.
+
+    :param module: Name of driver module to load.
+    :param classname: Name of driver class to instantiate.
+    :param task: Name of driver task to execute.
+    :param schema_file: Path to schema file.
+    :param module_dir: Path to directory containing driver module.
+    :param config: Path to config file (read stdin if missing or None).
+    :param cycle: The cycle.
+    :param leadtime: The leadtime.
+    :param batch: Submit run to the batch system?
+    :param dry_run: Do not run the executable, just report what would have been done.
+    :param graph_file: Write Graphviz DOT output here.
+    :param key_path: Path of keys to subsection of config file.
+    :param stdin_ok: OK to read from stdin?
+    :return: ``True`` if task completes without raising an exception.
+    """
+    if not (class_ := _get_driver_class(classname=classname, module=module, module_dir=module_dir)):
+        return False
+    accepted = set(getfullargspec(class_).args)
+    non_optional = {"cycle", "leadtime"}
+    required = accepted & non_optional
+    kwargs = dict(
+        config=ensure_data_source(config, bool(stdin_ok)),
+        dry_run=dry_run,
+        key_path=key_path,
+        schema_file=schema_file,
+    )
+    for arg in sorted(non_optional):
+        if arg in accepted and locals()[arg] is None:
+            log.error("%s requires argument '%s'", classname, arg)
+            return False
+    for arg in sorted(["batch", *required]):
+        if arg in accepted:
+            kwargs[arg] = locals()[arg]
+    driverobj = class_(**kwargs)
+    log.debug("Instantiated %s with: %s", classname, kwargs)
+    getattr(driverobj, task)()
+    if graph_file:
+        with open(graph_file, "w", encoding="utf-8") as f:
+            print(graph(), file=f)
+    return True
+
+
+def tasks(
+    classname: str,
+    module: str,
+    module_dir: Optional[Union[Path, str]] = None,
+) -> dict[str, str]:
+    """
+    Returns a mapping from task names to their one-line descriptions.
+
+    :param classname: Class of driver object to instantiate.
```

Comment:
```suggestion
    :param classname: Name of driver class to instantiate.
```

Consistency with other docstrings in this module.